common/src/model_registration/register.py

```diff
@@ -152,7 +150,10 @@ def main(args):
     # Updating tags and properties with value provided in metadata file
     if args.model_metadata:
         with open(args.model_metadata, "r") as stream:
-            metadata = json.load(stream)
+            try:
+                metadata = json.load(stream)
+            except:
+                metadata = yaml.safe_load(stream)
```

Comment:
I don't think this will work without reopening the stream or maybe trying what's recommended here: https://stackoverflow.com/questions/2106820/re-open-files-in-python

I don't know if the json/yaml modules will support random access, though.